src/protocol/context/malicious.rs

```diff
@@ -0,0 +1,101 @@
+use crate::ff::Field;
+use crate::helpers::messaging::{Gateway, Mesh};
+use crate::helpers::Role;
+use crate::protocol::context::{ContextInner, ProtocolContext, SemiHonestProtocolContext};
+use crate::protocol::malicious::SecurityValidatorAccumulator;
+use crate::protocol::prss::{
+    Endpoint as PrssEndpoint, IndexedSharedRandomness, SequentialSharedRandomness,
+};
+use crate::protocol::{Step, Substep};
+use crate::secret_sharing::{MaliciousReplicated, Replicated};
+use std::sync::Arc;
+
+/// Represents protocol context in malicious setting, i.e. secure against one active adversary
+/// in 3 party MPC ring.
+#[derive(Clone, Debug)]
+pub struct MaliciousProtocolContext<'a, F: Field> {
+    inner: ContextInner<'a>,
```

Comment:
@martinthomson I remember we were talking about using references here, but I couldn't find a good way to do it. there must be something that holds onto this inner struct for the duration of protocol execution. It didn't seem very ergonomic but I could be missing something

Reply:
I think the idea was to collect the static components of the context (role, prss, gateway) into `GlobalContext` so that a single `&GlobalContext` needs to be duplicated each time the context is refined, rather than multiple items.

This is something that I was planning to look at, but I haven't done anything with it yet. Hopefully it can be done in a way that doesn't change the context API.

Reply:
With #261 I think we can make it ergonomic to use because we can just stitch global contexts inside test world and create contexts on demand. For real world implementation I am not sure if we can make it better than just holding onto both global context and protocol context